cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -283,23 +285,23 @@ private long processMessages(String topic, int partition,
         // Although it's not the same as the in memory object size, it should be just a constant factor, hence
         // it is proportional to the actual object size.
         eventQueue.add(loggingEvent, loggingEvent.getTimeStamp(),
-                       message.message().payloadSize(), partition, message.offset());
+                       message.message().payloadSize(), partition, message.nextOffset());
```

Comment:
Later we need to persist the smallest offset and corresponding timestamp of the unprocessed events in the queue. If we add `message.nextOffset()` here, we will persist the nextOffset of the smallest offset instead.